src/serac/physics/materials/solid_functional_material.hpp

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file solid_functional_material.hpp
+ *
+ * @brief The material and load types for the solid functional physics module
+ */
+
+#pragma once
+
+#include "serac/numerics/functional/functional.hpp"
+
+/// SolidFunctional helper data types
+namespace serac::solid_util {
+
+/**
+ * @brief Linear isotropic elasticity material model
+ * 
+ * @tparam dim Spatial dimension of the mesh
+ */
+template <int dim>
+class LinearIsotropicElasticity {
+public:
+  /**
+   * @brief Construct a new Linear Isotropic Elasticity object
+   * 
+   * @param density Density of the material
+   * @param shear_modulus Shear modulus of the material
+   * @param bulk_modulus Bulk modulus of the material
+   */
+  LinearIsotropicElasticity(double density = 1.0, double shear_modulus = 1.0, double bulk_modulus = 1.0)
+      : density_(density), bulk_modulus_(bulk_modulus), shear_modulus_(shear_modulus)
+  {
+    SLIC_ERROR_ROOT_IF(shear_modulus_ < 0.0,
+                       "Shear modulus must be positive in the linear isotropic elasticity material model.");
+
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic elasticity material model.");
+
+    SLIC_ERROR_ROOT_IF(bulk_modulus_ < 0.0,
+                       "Bulk modulus must be positive in the linear isotropic elasticity material model.");
```

Comment:
```suggestion
    SLIC_ERROR_ROOT_IF(shear_modulus_ < 0.0,
                       "Shear modulus must be positive in the linear isotropic elasticity material model.");

    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic elasticity material model.");

    SLIC_ERROR_ROOT_IF(bulk_modulus_ < 0.0,
                       "Bulk modulus must be positive in the linear isotropic elasticity material model.");
                       
    double K = bulk_modulus;
    double G = shear_modulus;
    double poisson_ratio = (3 * K - 2 * G) / (6 * K + 2 * G));
  
    SLIC_ERROR_ROOT_IF(poisson_ratio < 0.0,
                       "Poisson ratio must be positive in the linear isotropic elasticity material model.");
                       
```
see: https://github.com/LLNL/serac/issues/586#issuecomment-954365299